### PR TITLE
improve robustness to invalid data

### PR DESCRIPTION
**Pull request by andywilkinshmcts**
https://tools.hmcts.net/jira/browse/EXUI-1117

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/complex/read-complex-field.component.ts [hunk 1]
```diff
@@ -30,8 +30,8 @@ export class ReadComplexFieldComponent extends AbstractFieldReadComponent implem
           field.list_items = this.caseField.value[field.id].list_items;
           field.value = {
             list_items: field.list_items,
-            value: this.caseField.value[field.id].value && this.caseField.value[field.id].value.code ?
-                    this.caseField.value[field.id].value.code :
+            value: this.caseField.value[field.id].value && this.caseField.value[field.id].value?.code ?
```
Comment (Daniel-Lam) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/complex/read-complex-field.component.ts:33]: @andywilkinshmcts - if you're doing `this.caseField.value[field.id].value?`, then the previous `this.caseField.value[field.id].value` is redundant

**Review by Daniel-Lam — DISMISSED** [review]
@andywilkinshmcts - just one comment that needs addressing

**Review by connorpgpmcelroy — APPROVED** [review]
LGTM

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/dynamic-radio-list/write-dynamic-radio-list-field.component.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ export class WriteDynamicRadioListFieldComponent extends AbstractFieldWriteCompo
     }
 
     if (!this.caseField.value && this.caseField.formatted_value && this.caseField.formatted_value.value) {
```
Comment (olusegz07) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/dynamic-radio-list/write-dynamic-radio-list-field.component.ts:21]: The if statement can be further simplified - if (!this.caseField?.value && this.caseField?.formatted_value?.value) { }.

**Review by RiteshHMCTS — APPROVED** [review]

**Review by Josh-HMCTS — APPROVED** [review]

**Pull request merged**